torch/_dynamo/symbolic_convert.py

```diff
@@ -2207,13 +2207,30 @@ def check_inlineable(func):
 
             # _origin marks this as coming from an internal dynamo known function that is safe to
             # trace through.
-            if hasattr(func.fn, "_origin") and func.fn._origin in [
+            if hasattr(func, "fn") and hasattr(func.fn, "_origin") and func.fn._origin in [
                 produce_trampoline_autograd_fwd,
                 produce_trampoline_autograd_apply,
                 produce_trampoline_autograd_bwd,
             ]:
                 # Known sound
                 return
+
+
+            # Function returned by torch.autograd.function.once_differentiable is known sound
+            if (
+                isinstance(func, UserFunctionVariable) and
+                func.get_filename().endswith('torch/autograd/function.py') and
+                not func.fn.__module__.startswith('torch.autograd.function')
+            ):
+                return
+            # Function wrapped by torch.autograd.function.once_differentiable is known sound
+            if (
+                isinstance(func, NestedUserFunctionVariable) and
+                func.get_filename().endswith('torch/autograd/function.py') and
+                func.fn_name.value == 'once_differentiable.<locals>.wrapper.<locals>.<genexpr>'
```

Comment:
Seems super hacky. I can try the "decorator to mark a function to always be inlined" idea if we think this is too hacky.

Reply:
Yeah as this is, it doesn't spark joy.

One thing we can do is like `wrapper.once_differentiable = True` and then just hasattr check that field? Still gross, but beats this value check on string (which is not safe anyway, if someone refactors once_differentiable lol)

What was wrong with inlining btw?

Reply:
> What was wrong with inlining btw?

Before this PR, Dynamo attempts to inline the function produced by once_differentiable. Then, it fails in `InliningInstructionTranslator.check_inlineable` and complains that the function being inlined is in skipfiles.